### DISCUSSION TITLE
Getting Error while executing Example.py

### DISCUSSION THREAD

**Discussion by sohanrao123** [TechfaneTechnologies/QtsApp#8] [Q&A] [answered]
Hello, the error i am facing is issue when executing example.py file.
![image](https://user-images.githubusercontent.com/60308272/187029317-30089345-8f78-452b-ad74-745418299910.png)

can you please suggest what other things need to installed prior to running this script?

**Comment by MooneDrJune** [comment, marked as answer]
> Hello, the error i am facing is issue when executing example.py file. ![image](https://user-images.githubusercontent.com/60308272/187029317-30089345-8f78-452b-ad74-745418299910.png)
> 
> can you please suggest what other things need to installed prior to running this script?

The error is arising, due to the reason that you don't have `requests` module installed in your python program.

To install the same run this command `pip install -U requests`

**Reply by sohanrao123** [reply]
Hello @MooneDrJune , the problem was resolved, but upon executing `python ./example.py` I get this following error, and the excel sheet doesn't update at all, can you please suggest what needs to be done?
![image](https://user-images.githubusercontent.com/60308272/187124394-e3492c13-42c0-40ad-8238-43ae404cef56.png)

**Reply by MooneDrJune** [reply]
Refer this [issue ](https://github.com/TechfaneTechnologies/QtsApp/issues/6),
You have to complete your profile at qtsapp website.

**Comment by dgsalian** [comment]
![image](https://user-images.githubusercontent.com/45466045/201001085-4c37b425-9ded-4103-bcbf-a3ee97b58ec9.png)
data not updating for CallOI change and Call OI rest is working.

**Reply by MooneDrJune** [reply]
OI & change in OI will be updated every 3 minutes.

**Reply by dgsalian** [reply]
> OI & change in OI will be updated every 3 minutes.

Yes I understand Exchange release OI every 3 min but in this sheet only this data not updating like now it has been updated 30 min back

**Reply by dgsalian** [reply]
![image](https://user-images.githubusercontent.com/45466045/201023419-5acb18b9-25cd-41ec-9a17-12f5640e2758.png)

**Reply by MooneDrJune** [reply]
> ![image](https://user-images.githubusercontent.com/45466045/201023419-5acb18b9-25cd-41ec-9a17-12f5640e2758.png)

Do one thing, change this https://github.com/TechfaneTechnologies/QtsApp/blob/1f9156c539c6ebe16d7ff0224e8373ed292779b4/example.py#L30

to 

`stream = False`

and rerun the program and update here, whether its updating or not ?

**Reply by dgsalian** [reply]
![image](https://user-images.githubusercontent.com/45466045/201055812-41d648ee-641b-4fda-a3d9-ed026ef1a553.png)

**Comment by dkmevawala** [comment]
![image](https://user-images.githubusercontent.com/81729834/203023340-7c71b875-ba31-4bb3-961a-e079c2a18710.png)
data not getting updated ..showing data of 10th Aug
![image](https://user-images.githubusercontent.com/81729834/203023441-8e67789e-c12b-4848-8206-a282ba60c6b7.png)

**Comment by yashpal2002** [comment]
File "C:\Users\hp\AppData\Local\Programs\Python\Python310\lib\site-packages\qtsapp\lib.py", line 41, in <module>
    from websocket import create_connection, _exceptions
ImportError: cannot import name 'create_connection' from 'websocket' (C:\Users\hp\AppData\Local\Programs\Python\Python310\lib\site-packages\websocket\__init__.py)

**Reply by ShabbirHasan1** [reply]
> File "C:\Users\hp\AppData\Local\Programs\Python\Python310\lib\site-packages\qtsapp\lib.py", line 41, in from websocket import create_connection, _exceptions ImportError: cannot import name 'create_connection' from 'websocket' (C:\Users\hp\AppData\Local\Programs\Python\Python310\lib\site-packages\websocket__init__.py)

`pip install --upgrade websocket websocket-client`

**Comment by VadivelanTEST** [comment]
Traceback (most recent call last):
  File "C:\wamp64\www\zip\hlo\QtsApp\example.py", line 32, in <module>
    EnchantQtsApp(
  File "C:\Users\Eagleminds\AppData\Local\Programs\Python\Python311\Lib\site-packages\qtsapp\QtsApp.py", line 158, in EnchantQtsApp
    del qtsapp
        ^^^^^^
UnboundLocalError: cannot access local variable 'qtsapp' where it is not associated with a value

**Comment by ketan680** [comment]
plz help me. 
Plz send process run program
![Untitled](https://github.com/TechfaneTechnologies/QtsApp/assets/147753668/1ce35731-0805-471d-ab46-96ea6c7e8223)
